distribution/sender/routes/database-event-reader.xml

```diff
@@ -0,0 +1,115 @@
+<routes xmlns="http://camel.apache.org/schema/spring">
+
+    <route id="db-event-reader">
+        <from uri="scheduler:retry?initialDelay=1000&amp;delay=10000" />
+  
+		<log message="Fetching events in the staging table" loggingLevel="DEBUG" />
+    
+        <toD uri="jpa:DebeziumEvent?query=SELECT dE.id FROM DebeziumEvent dE ORDER BY event_time_stamp ASC" />
+			
+			
+       	 <choice>
+            <when>
+                <simple>${body.size()} > 0</simple>
+                <log message="Event count in the stating area: ${body.size()}" />
+                 
+                 <split parallelProcessing="false" stopOnException="true">
+                 	<simple>${body}</simple>
+                 	
+                	<log message="Loading event item with id: ${body}" loggingLevel="INFO"/>
+                    
+                    <setProperty name="debezium-event-id">
+                        <simple>${body}</simple>
+                    </setProperty>
+                    
+                    <toD uri="jpa:DebeziumEvent?query=SELECT dE FROM DebeziumEvent dE WHERE dE.id = ${body}" />
+
+                    <setProperty name="event">
+                        <simple>${body[0].event}</simple>
+                    </setProperty>
+	                 
+                 	<doTry>
+                 		<setProperty name="is-operation-completed">
+				            <spel>false</spel>
+				        </setProperty>
+				        
+	                    <log message="Loaded event from staging table: ${exchangeProperty.event}" loggingLevel="INFO" />
+	                     
+                        <setProperty name="route-retry-count-map">
+	                    	<spel>#{new java.util.HashMap()}</spel>
+	                	</setProperty>
+	                	
+		                <split>
+		                    <simple>{{db-event.destinations}}</simple>
```

Comment:
What is this code for? I think you just copied and pasted logic from the retry route, please remove all this unused code

Reply:
this reader route should just read the event and call the db-event-listener route

Reply:
You mean db-event-processor??